numba/tests/test_np_functions.py

```diff
@@ -2768,6 +2772,66 @@ def test_repeat_exception(self):
                 with self.assertRaises(TypingError):
                     nbfunc(np.ones(1), rep)
 
+    def test_select(self):
+        x = np.arange(10)
+        np_pyfunc = np_select
+        np_nbfunc = njit(np_select)
+        for condlist,  choicelist, default in [
+            # Each test case below is one tuple.
+            # Each tuple is separated by a blank line
+            ([x < 3, x > 5], [x, x ** 2],0),
+
+            ((x < 3, x > 5), (x, x ** 2), 0),
+
+            ([np.array([False, False, False]),
+              np.array([False, True, False]),
+              np.array([False, False, True])],
+             [np.array([1, 2, 3]),
+              np.array([4, 5, 6]),
+              np.array([7, 8, 9])],15),
+
+            ([np.array([True]),
+              np.array([False])], [np.array([1]), np.array([2])], 0),
+
+            ([np.array([False])] * 100, [np.array([1])] * 100, 0)
```

Comment:
This is a good start to test cases, perhaps have a play around with NumPy and see what other things are accepted/rejected?

Reply:
@stuartarchibald, I added more tests. I have all the ones from Numpy plus a few more combinations, including complex numbers, NaNs, and 2d arrays. Also wrote down what each case is supposed to be testing.